docs/types/value-types.rst

```diff
@@ -494,21 +496,22 @@ regardless of the type of the right (exponent) operand.
     Division on integer literals used to truncate in Solidity prior to version 0.4.0, but it now converts into a rational number, i.e. ``5 / 2`` is not equal to ``2``, but to ``2.5``.
 
 .. note::
-    Solidity has a number literal type for each rational number.
-    Integer literals and rational number literals belong to number literal types.
+    In solidity, rational numbers and integers expressed as literals are of number literal type.
     Moreover, all number literal expressions (i.e. the expressions that
-    contain only number literals and operators) belong to number literal
-    types.  So the number literal expressions ``1 + 2`` and ``2 + 1`` both
-    belong to the same number literal type for the rational number three.
+    contain only number literals and operators) are of number literal
+    type.  The number literal expressions ``1 + 2`` and ``2 + 1`` are both
+    of number literal type, just the same as ``3``. Technically speaking, the internal representation
+    for fractional rational numbers like ``1.5`` is different than that of ``3``, specifically to support
+    arbitrary precision. However, all number literals are treated as effectively the same type.
 
 
 .. note::
     Number literal expressions are converted into a non-literal type as soon as they are used with non-literal
-    expressions. Disregarding types, the value of the expression assigned to ``b``
-    below evaluates to an integer. Because ``a`` is of type ``uint128``, the
-    expression ``2.5 + a`` has to have a proper type, though. Since there is no common type
-    for the type of ``2.5`` and ``uint128``, the Solidity compiler does not accept
-    this code.
+    expressions. If not for type conversion considerations, when ``a=1`` the value of the expression ``2.5 + a + 0.5``
+    would be the integer ``4``. However, ``a`` is of type ``uint128``. Therefore, an implicit conversion to a non-literal
```

Comment:
Technically, it could be a literal type if a common literal type existed.
```suggestion
    would be the integer ``4``. However, ``a`` is of type ``uint128``. Therefore, an implicit conversion to a common
```